### ISSUE TITLE
Wrong (+) to LEFT JOIN transformation in join tree A⟕B, A⟕C, A⋈D

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12324]
This query:

```sql
select *
from 
  t1 a,
  t2 b,
  t3 c,
  t4 d
where
  a.c1 = b.c1(+)
  and a.c1 = c.c1(+)
  and a.c1 = d.c1
```

Produces this wrong output when transforming Oracle joins to ANSI joins:

```sql
select *
from T1 A
  left outer join T2 B
    on A.C1 = B.C1
  left outer join T3 C
    on A.C1 = C.C1
  left outer join T4 D
    on A.C1 = B.C1
```

There are two problems with the `JOIN` of `T4`:

- A `LEFT JOIN` is generated, when it should be an `INNER JOIN`
- The wrong table alias `B.C1` is used to dereference `C1`, instead of `D.C1`.

The expected output query is:

```sql
select *
from T1 A
  left outer join T2 B
    on A.C1 = B.C1
  left outer join T3 C
    on A.C1 = C.C1
  join T4 D
    on A.C1 = D.C1
```

**Comment by lukaseder** [comment]
The problem doesn't appear if we switch the order of tables in the table list, e.g.

```sql
select *
from 
  t4 d,
  t1 a,
  t2 b,
  t3 c
where
  a.c1 = b.c1(+)
  and a.c1 = c.c1(+)
  and a.c1 = d.c1
```

Produces the correct result:

```sql
select *
from T4 D
  join (
    T1 A
      left outer join T2 B
        on A.C1 = B.C1
      left outer join T3 C
        on A.C1 = C.C1
  )
    on A.C1 = D.C1
```

And

```sql
select *
from 
  t1 a,
  t4 d,
  t2 b,
  t3 c
where
  a.c1 = b.c1(+)
  and a.c1 = c.c1(+)
  and a.c1 = d.c1
```

Produces

```sql
select *
from T1 A
  left outer join T2 B
    on A.C1 = B.C1
  left outer join T3 C
    on A.C1 = C.C1
  join T4 D
    on A.C1 = D.C1
```

**Comment by lukaseder** [comment]
And

```sql
select *
from 
  t1 a,
  t2 b,
  t4 d,
  t3 c
where
  a.c1 = b.c1(+)
  and a.c1 = c.c1(+)
  and a.c1 = d.c1
```

Produces

```sql
select *
from T1 A
  left outer join T3 C
    on A.C1 = C.C1
  join T4 D
    on A.C1 = D.C1
  left outer join T2 B
    on A.C1 = B.C1
```

**Comment by lukaseder** [comment]
This seems to be a regression introduced by https://github.com/jOOQ/jOOQ/issues/10697's preventing the re-ordering of tree structures, which worked well for A⟕B, A⋈C trees, but apparently not for A⟕B, A⟕C, A⋈D trees

**Comment by lukaseder** [comment]
The prevention of re-ordering is not applied (yet) when mixing inner and outer joins, so the result will be this:

```sql
select *
from T1 A
  join T4 D
    on A.C1 = D.C1
  left outer join T2 B
    on A.C1 = B.C1
  left outer join T3 C
    on A.C1 = C.C1
```

Which is less nice, but at least it's correct.

**Comment by lukaseder** [comment]
The re-ordering will be fixed separately: https://github.com/jOOQ/jOOQ/issues/11996

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12326)

**Comment by lukaseder** [comment]
This issue, which also depends on the table declaration order, seems related: https://github.com/jOOQ/jOOQ/issues/11692
